gensim/models/keyedvectors.py

```diff
@@ -311,11 +321,18 @@ def doesnt_match(self, words):
         """
         self.init_sims()
 
-        words = [word for word in words if word in self.vocab]  # filter out OOV words
-        logger.debug("using words %s" % words)
         if not words:
             raise ValueError("cannot select a word from an empty list")
-        vectors = vstack(self.syn0norm[self.vocab[word].index] for word in words).astype(REAL)
+        logger.debug("using words %s" % words)
```

Comment:
Use lazy log formatting (params passed via comma, not directly formatted, which is mostly wasteful because `debug` is not output).